Arrowgene.Ddon.GameServer/Handler/CraftStartQualityUpHandler.cs

```diff
@@ -0,0 +1,283 @@
+#nullable enable
+using System.Linq;
+using Arrowgene.Ddon.GameServer.Characters;
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Ddon.Shared.Network;
+using Arrowgene.Logging;
+using System;
+using System.Collections.Generic;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class CraftStartQualityUpHandler : GameStructurePacketHandler<C2SCraftStartQualityUpReq>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(CraftStartQualityUpHandler));
+        private static readonly List<StorageType> STORAGE_TYPES = new List<StorageType> {
+            StorageType.ItemBagConsumable, StorageType.ItemBagMaterial, StorageType.ItemBagEquipment, StorageType.ItemBagJob, 
+            StorageType.StorageBoxNormal, StorageType.StorageBoxExpansion, StorageType.StorageChest
+        };
+        private static readonly List<StorageType> StorageEquipNBox = new List<StorageType> {
+            StorageType.ItemBagEquipment, StorageType.StorageBoxNormal, StorageType.StorageBoxExpansion, StorageType.StorageChest, StorageType.CharacterEquipment
+        };
+
+        private readonly ItemManager _itemManager;
+        private readonly Random _random;
+
+        public CraftStartQualityUpHandler(DdonGameServer server) : base(server)
+        {
+            _itemManager = Server.ItemManager;
+            _random = new Random();
+        }
+
+        public override void Handle(GameClient client, StructurePacket<C2SCraftStartQualityUpReq> packet)
+        {
+
+            S2CItemUpdateCharacterItemNtc updateCharacterItemNtc = new S2CItemUpdateCharacterItemNtc();    
```

Comment:
Same comment as other file. Refactor this function to make it more readable (it is too many lines).